lib/teleterm/api/proto/v1/access_request.proto

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.terminal.v1;
+
+option go_package = "github.com/gravitational/teleport/lib/teleterm/v1";
+
+message AccessRequest {
+  string id = 1;
+  string state = 2;
```

Comment:
Any reason this is a string an not an enum like in types.proto?

I haven't used proto enums too much in JS yet so idk, perhaps they're not that ergonomic to use. OTOH we use TypeScript there so I feel like we should be able to convert it nicely there (or maybe the generator would do that for us).

Reply:
If we're to keep this as a string, it'd be worthwhile to list possible options here but at that point we could use an enum.

Reply:
Yeah, now that I think back on it, it's a bit weird as its a string here but a RequestState in the Create. I believe it has to do with the ergonomics as you said. Setting something to a specific enum like RequestState.PENDING is a lot easier than mapping the other way around during read (at least from my experience). Also, the [web ui sends AccessRequests](https://github.com/gravitational/teleport.e/blob/7ee4f8b44071d13e311c29761a81c7d2fb5a0bad/lib/web/ui/access_request.go#L11-L42) as a string which is where I got the idea. I think i'll list the possible states here as comment and leave as string since this is during read and mutate will keep the enum.